exercises/alphametics/test/Tests.hs

```diff
@@ -0,0 +1,76 @@
+import Test.Hspec.Runner (configFastFail, defaultConfig, hspecWith)
+import Data.List (sort)
+
+import Alphametics (solve)
+
+main :: IO ()
+main = hspecWith defaultConfig {configFastFail = True} specs
+
+specs :: Spec
+specs = describe "alphametics" $
+          describe "solve" $ for_ cases test
+  where
+
+    test Case{..} = it description assertion
+      where
+        assertion  = solution `shouldBe` sort <$> expected
+        solution = sort <$> solve puzzle
+
+-- Test cases adapted from `exercism/x-common/exercises/alphametics/acanonical-data.json` on 2016-10-26.
+
+data Case = Case { description  ::        String
+                 , puzzle       ::        String
+                 , expected     ::        Maybe[(Char, Int)]
+                 }
+
+cases :: [Case]
+cases = [ Case { description  = "puzzle with three letters"
+               , puzzle       = "I + BB == ILL"
+               , expected     = Just [('I', 1), ('B', 9), ('L', 0)]
+               }
+        , Case { description  = "solution must have unique value for each letter"
+               , puzzle       = "A == B"
+               , expected     = Nothing
+               }
+        , Case { description  = "leading zero solution is invalid"
+               , puzzle       = "ACA + DD == BD"
+               , expected     = Nothing
+               }
+        , Case { description  = "puzzle with four letters"
+               , puzzle       = "AS + A == MOM"
+               , expected     = Just [('A', 9), ('S', 2), ('M', 1), ('O', 0)]
+               }
+        , Case { description  = "puzzle with six letters"
+               , puzzle       = "NO + NO + TOO == LATE"
+               , expected     = Just [('N', 7), ('O', 4), ('T', 9), ('L', 1),
+                                      ('A', 0), ('E', 2)]
+               }
+        , Case { description  = "puzzle with seven letters"
+               , puzzle       = "HE + SEES + THE == LIGHT"
+               , expected     = Just [('E', 4), ('G', 2), ('H', 5), ('I', 0),
+                                      ('L', 1), ('S', 9), ('T', 7)]
+               }
+        , Case { description  = "puzzle with eight letters"
+               , puzzle       = "SEND + MORE == MONEY"
+               , expected     = Just [('S', 9), ('E', 5), ('N', 6), ('D', 7), ('M', 1),
+                                      ('O', 0), ('R', 8), ('Y', 2)]
+               }
+        , Case { description  = "puzzle with ten letters"
+               , puzzle       = "AND + A + STRONG + OFFENSE + AS + A + GOOD == DEFENSE"
+               , expected     = Just [('A', 5), ('D', 3), ('E', 4), ('F', 7), ('G', 8),
+                                      ('N', 0), ('O', 2), ('R', 1), ('S', 6), ('T', 9)]
+               }
+        , Case { description  = "puzzle with multiplication"
+               , puzzle       = "IF * DR == DORI"
+               , expected     = Just [('I', 8), ('F', 2), ('D', 3), ('R', 9), ('O', 1)]
```

Comment:
It could still be the case that many students will still submit the simple brute force solution anyway. It's the first thing that comes to mind. Different people want to see different things from Exercism. So limiting to addition problems opens the door to doing something more considered, for those who want to do that. The decreased difficult does help a bit too, I think.

I realize that another way to expanding this problem is indeed to keep the operations other than addition. There are various directions to take this problem. I chose one way (and it happens to be the way that x-common chose too), but maybe others like a different way.

As for what we can do about this long runtime, maybe we can get some ideas from other tracks. https://github.com/exercism/xruby/pull/455/files#diff-8ec059eaf4c70572282d3173a6325c3aR53 the Ruby track chose to comment out the cases that had many letters since they do take a long time to solve that way. Recently as of https://github.com/exercism/xruby/pull/469 the Ruby example is fast enough to solve the equations with many letters, but they still have a comment telling students it's OK not to do them: https://github.com/exercism/xruby/pull/469/files#diff-8ec059eaf4c70572282d3173a6325c3aR53 . If we decide here that the same ideas are appropriate, we could apply them.